Add socket stats API documentation (#841)
ARMmbed/mbed-os-5-docs: docs/api/networksocket/SocketStats.md

## SocketStats

Use the SocketStats class to read the network socket statistics. SocketStats Class object is created by "InternetSocket" class when statistics are enabled by `nsapi.socket-stats-enable` config option.

> SocketStats class provides static API `mbed_stats_socket_get_each` to get the network socket information. Using this API you can know how many sockets are open, destination IPand data bytes sent and received. Also the timestamp of recent "socket state" change is recorded in `last_change_tick`.

Reviewer: ```suggestion
The `SocketStats` class provides the static function `mbed_stats_socket_get_each` to get the network socket information for all sockets both currently in use and closed. Using this API you can know how many sockets are open, destination IP addresses, the amount of data sent and received in bytes, the type of socket and the timestamp of recent "socket state" changes recorded in `last_change_tick`.
```

> The `SocketStats` class provides the static function `mbed_stats_socket_get_each` to get the network socket information for all sockets both currently in use and closed. Using this API you can know how many sockets are open, destination IP addresses, the amount of data sent and received in bytes, the type of socket and the timestamp of recent "socket state" changes recorded in `last_change_tick`.

Reviewer 2: Please change "Using this API.." to "Using this function...". API usually refers to a set of functions/routines(courtesy: Kevin G). Everything else looks good to me.

Author: Done. 👍

> The `SocketStats` class provides the static function `mbed_stats_socket_get_each` to get the network socket information for all sockets both currently in use and closed. Using this API you can know how many sockets are open, destination IP addresses, the amount of data sent and received in bytes, the type of socket and the timestamp of recent "socket state" changes recorded in `last_change_tick`.

Author: ```suggestion
The `SocketStats` class provides the static function `mbed_stats_socket_get_each` to get the network socket information for all sockets both currently in use and closed. Using this function you can know how many sockets are open, destination IP addresses, the amount of data sent and received in bytes, the type of socket and the timestamp of recent "socket state" changes recorded in `last_change_tick`.
```

> Socket Statistics are stored in intenal data buffer and maintained for closed sockets as well. In case internal data buffer is full, socket closed with oldest timestamp will be over-written. Maximum amount of sockets statistics stored can be configured by `nsapi.socket-stats-max-count` config option.

Reviewer: Link to config page maybe @AnotherButler ?

Author: @kegilbert - I don't see network configs here https://github.com/ARMmbed/mbed-os-5-docs/tree/development/docs/reference/configuration

Reviewer: Good point, forgot to check for that earlier. We should add that page as well. Can do it in this PR or another one.

Author: Will be good to do it in another one.

### SocketStats class reference

> [![View code](https://www.mbed.com/embed/?type=library)](http://os-doc-builder.test.mbed.com/docs/development/mbed-os-api-doxy/)

Author: @AnotherButler - Network stats PR https://github.com/ARMmbed/mbed-os/pull/8592 is merged, and doxygen for new Class will be generated from features/netsocket/SocketStats.h. Please help me with the link here.

Reviewer 3: http://os-doc-builder.test.mbed.com/docs/development/mbed-os-api-doxy/class_socket_stats.html

> [![View code](https://www.mbed.com/embed/?type=library)](http://os-doc-builder.test.mbed.com/docs/development/mbed-os-api-doxy/)

Author: ```suggestion
[![View code](https://www.mbed.com/embed/?type=library)](http://os-doc-builder.test.mbed.com/docs/development/mbed-os-api-doxy/class_socket_stats.html)
```
